snapcraft/commands/init.py

```diff
@@ -16,74 +16,44 @@
 
 """Snapcraft init command."""
 
-from pathlib import Path
-from textwrap import dedent
+import argparse
 
-from craft_application.commands import AppCommand
-from craft_cli import emit
-from overrides import overrides
+import craft_application.commands
+import craft_cli
+from typing_extensions import override
 
 from snapcraft import errors
 from snapcraft.parts.yaml_utils import get_snap_project
 
-_TEMPLATE_YAML = dedent(
-    """\
-    name: my-snap-name # you probably want to 'snapcraft register <name>'
-    base: core24 # the base snap is the execution environment for this snap
-    version: '0.1' # just for humans, typically '1.2+git' or '1.3.2'
-    summary: Single-line elevator pitch for your amazing snap # 79 char long summary
-    description: |
-      This is my-snap's description. You have a paragraph or two to tell the
-      most important story about your snap. Keep it under 100 words though,
-      we live in tweetspace and your description wants to look good in the snap
-      store.
 
-    grade: devel # must be 'stable' to release into candidate/stable channels
-    confinement: devmode # use 'strict' once you have the right plugs and slots
+class InitCommand(craft_application.commands.InitCommand):
+    """Snapcraft init command."""
 
-    parts:
-      my-part:
-        # See 'snapcraft plugins'
-        plugin: nil
-    """
-)
+    @override
+    def run(self, parsed_args: argparse.Namespace) -> None:
+        """Pack a directory or run the lifecycle and pack all artifacts."""
+        project_dir = self._get_project_dir(parsed_args)
 
+        if parsed_args.name:
+            craft_cli.emit.progress(
+                "Ignoring '--name' parameter because it is not supported yet.",
```

Comment:
I'm not sure if this should be an error, a warning, or if I should change the default template to use this as the snap name.

Reply:
if you changed the default template to support the name, would you be able to drop this class wholesale?

Reply:
Alright. Then I think it should be an error and we should plan to address it soon

Reply:
`--name` is now supported and I've dropped the `InitClass` wholesale.